src/boat/private/Config.nim

```diff
@@ -1,10 +1,83 @@
 ##
 ## Config
 ## ======
-## base configuration for Manifest and CaptainsLog
+## provides a parser and standard interface
 
 ##[
 ## TLDR
-- a manifest eventually becomes the captains log
-  - from another perspective, the captains log is the realization of a manifest
+- come back later
+
+links
+-----
+- [parsecfg](https://nim-lang.org/docs/parsecfg.html)
+
 ]##
+
+import std/[
+  os,
+  parsecfg,
+  strutils,
+]
+
+import BoatErrors
+
+const manifestName = "manifest.nim.ini" ## \
+  ## the captains manifest must be named manifest.nim.ini
+  ## TODO: ensure this is reflected in the docs
+
+type Config* = ref object of RootObj
+  cachedir*: string ## \
+    ## defaults to getCacheDir();
+  use*: string ## \
+    ## path on disk pointing to a manifest / or dir with a manifest
+    ## URL pointing to a *.ini file
+  #--- private ---#
+  parsed: parsecfg.Config ## \
+    ## the parsed config after loading
+  saved: bool ## \
+    ## true if self.use has been saved to disk
+
+proc parseLocalManifest*(self: Config, path: string = self.use): bool =
+  ## parse self.use to self.parsed
+  ## prefer calling self.load or self.reload for validation
+  self.parsed = loadConfig path
+  result = true
+
+proc localManifestIsValid*(self: Config, path: string = self.use): bool =
+  ## throws if manifest not found, cant be read, or errors during parsing
+  let pathInfo = path.getFileInfo
+  result = case pathInfo.kind
+    of pcFile, pcLinkToFile:
+      if fpUserRead notin pathInfo.permissions: raise filePermissionError
+      elif not path.endsWith manifestName: raise manifestNameError
+      elif not self.parseLocalManifest path: raise configParseError
+      else: true
+    of pcDir, pcLinkToDir: self.localManifestIsValid self.use / manifestName
+
+proc save*(self: Config): bool =
+  ## serialize Self.parsed to disk @ self.cachedir | getCachDir() / <SELF.ID>.manifest.nim.ini
```

Comment:
the cachedir should be an implementation detail, and always set to getCacheDir()